application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelMonitor.java

```diff
@@ -0,0 +1,287 @@
+package org.togetherjava.tjbot.commands.free;
+
+import net.dv8tion.jda.api.entities.Category;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.GuildChannel;
+import net.dv8tion.jda.api.entities.TextChannel;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+
+/**
+ * A class responsible for monitoring the status of channels and reporting on their busy/free status
+ * for use by {@link FreeCommand}.
+ *
+ * Channels for monitoring are added via {@link #addChannelToMonitor(long)} however the monitoring
+ * will not be accessible/visible until a channel in the same {@link Guild} is registered for the
+ * output via {@link #addChannelForStatus(TextChannel)}. This will all happen automatically for any
+ * channels listed in {@link org.togetherjava.tjbot.config.FreeCommandConfig}.
+ *
+ * When a status channel is added for a guild, all monitored channels for that guild are tested and
+ * an {@link IllegalStateException} is thrown if any of them are not {@link TextChannel}s.
+ *
+ * After successful configuration, any changes in busy/free status will automatically be displayed
+ * in the configured {@code Status Channel} for that guild.
+ */
+public final class ChannelMonitor {
```

Comment:
is this class specifically designed for being used outside of the package? if not, maybe just make it package private.

Reply:
not sure yet
its not needed outside of free atm

but if others ever build other features that build on top of free command
(scoring system, times helpers are busiest telemetry, etc)
this is the class they will interact with

but at that point it will need a formal interface
i'll make it package private for now, we can always change it later.

Reply:
fixed